core/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -495,6 +510,20 @@ private ReducedQueryPhase reducedQueryPhase(Collection<? extends SearchPhaseResu
                 profileResults.put(key, result.consumeProfileResult());
             }
         }
+        
+        // AND the sets of unmapped fields across multiple indices.
+        Set<String> unanimouslyUnmappedFieldNames = null;
```

Comment:
maybe it would be simpler to just initialize `unanimouslyUnmappedFieldNames` with an `new HashSet`?

Reply:
We need to have a first-time-round special condition otherwise ANDing with `retainAll` on an initially empty set always produces an empty final set. I'm using ==null as the special first time condition. Alternative would be to init the set with first index but I'm not sure that's less code.